src/http.rs

```diff
@@ -135,6 +142,17 @@ where
 
         Box::pin(fut)
     }
+
+    fn maybe_add_proxy_headers(&self, request: &mut Request<B>) {
+        if let Some(proxy_headers) = self.proxy_connector.http_headers(request.uri()) {
+            for (k, v) in proxy_headers {
+                let request_headers = request.headers_mut();
```

Comment:
I suggest moving this out of the loop, to avoid calling it repeatedly.

Reply:
Absolutely, my oversight.